test/common/tracing/zipkin/zipkin_core_types_test.cc

```diff
@@ -461,6 +506,59 @@ TEST(ZipkinCoreTypesSpanTest, copyConstructor) {
   EXPECT_EQ(span.isSetTraceIdHigh(), span2.isSetTraceIdHigh());
 }
 
+TEST(ZipkinCoreTypesSpanTest, moveConstructor) {
+  Span span;
+
+  const uint64_t id = Util::generateRandom64();
+  const std::string id_hex = Hex::uint64ToHex(id);
+  span.setId(id);
+  span.setParentId(id);
+  span.setTraceId(id);
+  const int64_t timestamp =
+      std::chrono::duration_cast<std::chrono::microseconds>(
+          ProdSystemTimeSource::instance_.currentTime().time_since_epoch()).count();
+  span.setTimestamp(timestamp);
+  span.setDuration(3000LL);
+  span.setName("span_name");
+
+  Endpoint endpoint;
+  Annotation ann;
+  BinaryAnnotation bann;
+  std::vector<Zipkin::Annotation> annotations;
+
+  endpoint.setServiceName("my_service_name");
+  Network::Address::InstanceConstSharedPtr addr =
+      Network::Address::parseInternetAddressAndPort("192.168.1.2:3306");
+  endpoint.setAddress(addr);
+
+  ann.setValue(Zipkin::ZipkinCoreConstants::get().CLIENT_SEND);
+  ann.setTimestamp(timestamp);
+  ann.setEndpoint(endpoint);
+  annotations.push_back(ann);
+
+  span.setAnnotations(annotations);
+
+  Span span2(std::move(span));
+
+  EXPECT_EQ(id, span2.id());
```

Comment:
I don't think that is the case. My point is: there is no move in the literal sense. I believe what actually happens is some sort of "optimized copy". In fact, the data members of the original span seem to be intact.